### ISSUE TITLE
Fix KZG for identity group element

### ISSUE THREAD

**Issue by Rumata888** [AztecProtocol/barretenberg#79]
Currently gemini breaks for a polynomial, where the second half of coefficients is zero.

**Comment by adr1anh** [comment]
I don't think this is a problem with Gemini, but rather with KZG. At the end of Gemini, the last polynomials are constant, which is technically correct. Moreover, since we are batching all the opening with Shplonk, we don't actually compute quotients of these constant polynomials. 

My intuition is that it should work now already if we pass a half empty polynomial (though this is a good edge-case that we should test for in Sumcheck as well).

The issue that was coming up today is with the ZKG verifier. The KZG prover is constructing the quotient of a constant polynomial, which is the zero polynomial, so its commitment is the identity group element. This is completely valid. However, when the verifier performs the pairing check, the point at infinity causes panic.
